types/entrytimestamp_test.go

```diff
@@ -0,0 +1,182 @@
+		desc    string
+		in      []byte
+		want    EntryTimestampV1
+		wantErr string
+	}{
+		{
+			desc: "EmptyEntry",
+			in:   dehex("0001" + "0000000000000100" + "0000" + "0000"),
+			want: EntryTimestampV1{TimestampNanos: 0x100, LeafIdentityHash: []byte{}, MerkleLeafHash: []byte{}},
+		},
+		{
+			desc: "ValidEntry",
+			in:   dehex("0001" + "0000000000000100" + "0002" + "1011" + "0003" + "010305"),
+			want: EntryTimestampV1{TimestampNanos: 0x100, LeafIdentityHash: []byte{0x10, 0x11}, MerkleLeafHash: []byte{0x01, 0x03, 0x05}},
+		},
+		{
+			desc:    "WrongVersion",
+			in:      dehex("0000" + "0000000000000100" + "0000" + "0000"),
+			wantErr: "invalid EntryTimestamp.Version",
+		},
+		{
+			desc:    "TrailingData",
+			in:      dehex("0001" + "0000000000000100" + "0000" + "0000" + "AA"),
+			wantErr: "trailing data",
+		},
+		{
+			desc:    "IncompleteData",
+			in:      dehex("0001" + "0000"),
+			wantErr: "syntax error",
+		},
+		{
+			desc:    "NilInput",
+			in:      nil,
+			wantErr: "nil entry timestamp",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			var got EntryTimestampV1
+			err := got.UnmarshalBinary(test.in)
+			if err != nil {
+				if test.wantErr == "" {
+					t.Fatalf("UnmarshalBinary(%x)=nil,%q; want _,nil", test.in, err)
+				} else if !strings.Contains(err.Error(), test.wantErr) {
+					t.Fatalf("UnmarshalBinary(%x)=nil,%q; want nil, err containing %q", test.in, err, test.wantErr)
+				}
+				return
+			}
+			if test.wantErr != "" {
+				t.Fatalf("UnmarshalBinary(%x)=%+v,nil; want nil, err containing %q", test.in, got, test.wantErr)
+			}
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("UnmarshalBinary(%x)=%+v; want %+v", test.in, got, test.want)
+			}
+		})
+	}
+}
+
+func TestEntryDataForLeaf(t *testing.T) {
+	var tests = []struct {
+		desc    string
+		in      trillian.LogLeaf
+		want    []byte
+		wantErr string
+	}{
+		{
+			desc: "EmptyLeaf",
+			in: trillian.LogLeaf{
+				QueueTimestamp: &timestamppb.Timestamp{},
+			},
+			want: dehex("0001" + "0000000000000000" + "0000" + "0000"),
+		},
+		{
+			desc: "ValidEntry",
+			in: trillian.LogLeaf{
+				QueueTimestamp:   &timestamppb.Timestamp{Nanos: 0x100},
+				LeafIdentityHash: []byte{0x10, 0x11},
+				MerkleLeafHash:   []byte{0x01, 0x03, 0x05},
+			},
+			want: dehex("0001" + "0000000000000100" + "0002" + "1011" + "0003" + "010305"),
+		},
+		{
+			desc: "InvalidTimestamp",
+			in: trillian.LogLeaf{
+				QueueTimestamp:   &timestamppb.Timestamp{Seconds: 1, Nanos: -1},
+				LeafIdentityHash: []byte{0x10, 0x11},
+				MerkleLeafHash:   []byte{0x01, 0x03, 0x05},
+			},
+			wantErr: "failed to parse timestamp",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			got, err := EntryDataForLeaf(&test.in)
+			if err != nil {
+				if test.wantErr == "" {
+					t.Fatalf("EntryDataForLeaf(%+v)=nil,%q; want _,nil", test.in, err)
+				} else if !strings.Contains(err.Error(), test.wantErr) {
```

Comment:
No need for the `else` here since you're `Fatalf`'ing above (would then be consistent with L166 etc.)

Reply:
Left as-is, see previous discussion.